build/build-cdt-lib.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs');
+
+const outDir = `${__dirname}/../lighthouse-core/lib/cdt/generated`;
+const files = {
+  'node_modules/chrome-devtools-frontend/front_end/sdk/SourceMap.js': 'SourceMap.js',
+};
+
+// eslint-disable-next-line no-console
+console.log('making modifications ...');
+
+for (const [input, output] of Object.entries(files)) {
+  const code = fs.readFileSync(input, 'utf-8');
+
+  let lines = code.match(/^.*(\r?\n|$)/mg) || [];
+  const cutoffIndex = lines.findIndex(line => line.includes('Legacy exported object'));
+  lines = lines.splice(0, cutoffIndex);
+
+  let deletionMode = false;
+  let deletionCounter = 0;
+
+  const modifiedLines = lines.map((line, i) => {
+    // Don't modify jsdoc comments.
+    if (/^\s*[/*]/.test(line)) {
+      return line;
+    }
+    let newLine = line;
+
+    if (input.endsWith('SourceMap.js')) {
+      if (line.includes('static load(')) deletionMode = true;
+      if (line.includes('sourceContentProvider(')) deletionMode = true;
+      if (line.includes('Common.UIString')) newLine = '';
```

Comment:
it's removal didn't change my tests - I suppose it would matter if I used more of the methods in `TextSourceMap`? Since I didn't understand its usage I just deleted it.

if we truly wanted w/e this does, we'd need to consider our own i18n.

Reply:
i guess it does this: if a source file is the same as the compiled url (resolving with `new URL` helps get the source file correct), _and_ there's a sources content, the entry (when using `.mappings`) sourceURL will have `? [sm]` appended.

I don't want that, because the sizing function uses `sourceURL` as a key in its results.

Reply:
@patrickhulce - the build file is more targeted now.

@paulirish - see my comments in the code for why this should be removed.